src/main/java/com/nirima/jenkins/plugins/docker/DockerSlave.java

```diff
@@ -34,7 +34,7 @@
     private transient Run theRun;
 
     @DataBoundConstructor
-    public DockerSlave(DockerTemplate dockerTemplate, String containerId, String name, String nodeDescription, String remoteFS, int numExecutors, Mode mode, String labelString, ComputerLauncher launcher, RetentionStrategy retentionStrategy, List<? extends NodeProperty<?>> nodeProperties) throws Descriptor.FormException, IOException {
+    public DockerSlave(DockerTemplate dockerTemplate, String containerId, String name, String nodeDescription, String remoteFS, int numExecutors, Mode mode, int memoryLimit, int cpuShares, String labelString, ComputerLauncher launcher, RetentionStrategy retentionStrategy, List<? extends NodeProperty<?>> nodeProperties) throws Descriptor.FormException, IOException {
```

Comment:
You added `int memoryLimit, int cpuShares` as constructor arguments even though your aren't using them. Also they are `int` instead of `Integer` which will lead to NPE in `DockerTemplate` line 264 if the values are `null`.